pandas/core/groupby/generic.py

```diff
@@ -174,18 +175,36 @@ def _cython_agg_blocks(self, how, alt=None, numeric_only=True, min_count=-1):
 
                 # call our grouper again with only this block
                 obj = self.obj[data.items[locs]]
-                s = groupby(obj, self.grouper)
+                s = obj.groupby(self.grouper)
                 try:
                     result = s.aggregate(lambda x: alt(x, axis=self.axis))
                 except TypeError:
                     # we may have an exception in trying to aggregate
                     # continue and exclude the block
                     deleted_items.append(locs)
                     continue
+
+                if is_object_dtype(block.dtype) and how in ["prod", "cumprod", "sum"]:
```

Comment:
a way around is to have .aggregate different exceptions for flow control (make some private ones that inherit from NotImplementedError); we do this in the cython aggregator now (well we raise NotImplementedError)

Reply:
yah, distinguishing between our NotImplementedError vs numpy's is basically what #28198 does (albeit that is only in one place).  I think the groupby catching/raising is complicated enough that we need to do a few passes clearing up what gets caught where, then go through to do the catching in fewer places